gitlab/v4/objects/container_registry.py

```diff
@@ -49,6 +49,6 @@ def delete_in_bulk(self, name_regex=".*", **kwargs):
             GitlabDeleteError: If the server cannot perform the request
         """
         valid_attrs = ["keep_n", "older_than"]
-        data = {"name_regex": name_regex}
+        data = {"name_regex_delete": name_regex}
```

Comment:
We should probably keep our function arguments in sync with the API. I'd say either replace the `name_regex` variable to match the API (making this a breaking change) and we can add this to the 3.0.0 milestone (see also https://github.com/python-gitlab/python-gitlab/issues/1520), or fetch the old `name_regex` from kwargs if you want to keep backwards compatibility.

I'd prefer making a clean break with this if you're ok waiting for the 3.0.0 release for this to land.

Reply:
Yes, I agree. I was bit hesitant to make a breaking change. Third option would be to support both `name_regex` and `name_regex_delete` but that would just add complexity for very little gain.

I'll push the breaking change :+1:

Reply:
> We should probably keep our function arguments in sync with the API.

Do we want to keep the default values in sync as well? The old `name_regex` has default value of `.*` that is reflected in the function signature. However the new `name_regex_delete` does *not* have a default. Should the default be left as-is or should the argument be made required/non-optional?

The newly pushed commit (db6eed0) does the former.

Reply:
Hey @antti-mikael, thanks for keeping this PR up to date.

We now have 3.0.0 coming with the next scheduled release, so I think we can go with the breaking change.

I was thinking that we could drop the default value to follow the new API and make it required.

For reference, in the GUI it is also required now - you also have to enter a value even though the frontend provides a "default" grayed-out text (which will likely also be removed: https://gitlab.com/gitlab-org/gitlab/-/issues/333905#note_604415045).

Would you agree @JohnVillalovos?

Reply:
@nejch Sounds good to me. Thanks.

Reply:
> thanks for keeping this PR up to date

No problem.

> I was thinking that we could drop the default value to follow the new API and make it required.

Agreed, and commit 1099e96 should implement that change.